src/tests/AabbTest.cpp

```diff
@@ -107,6 +113,97 @@ void GeoTest::aabb() {
                          Cr::TestSuite::Compare::around(Mn::Vector3{eps}));
   }
 }
+
+void GeoTest::obbConstruction() {
+  OBB obb1;
+  const vec3f center(0, 0, 0);
+  const vec3f dimensions(20, 2, 10);
+  const quatf rot1 = quatf::FromTwoVectors(vec3f::UnitY(), vec3f::UnitZ());
+  OBB obb2(center, dimensions, rot1);
+
+  CORRADE_VERIFY(obb2.center().isApprox(center));
+  CORRADE_VERIFY(obb2.sizes().isApprox(dimensions));
+  CORRADE_VERIFY(obb2.halfExtents().isApprox(dimensions / 2));
+  CORRADE_VERIFY(obb2.rotation().coeffs().isApprox(rot1.coeffs()));
+
+  box3f aabb(vec3f(-1, -2, -3), vec3f(3, 2, 1));
+  OBB obb3(aabb);
+  CORRADE_VERIFY(obb3.center().isApprox(aabb.center()));
+  CORRADE_VERIFY(obb3.toAABB().isApprox(aabb));
+}
+
+void GeoTest::obbFunctions() {
+  const vec3f center(0, 0, 0);
+  const vec3f dimensions(20, 2, 10);
+  const quatf rot1 = quatf::FromTwoVectors(vec3f::UnitY(), vec3f::UnitZ());
+  OBB obb2(center, dimensions, rot1);
+  CORRADE_VERIFY(obb2.contains(vec3f(0, 0, 0)));
+  CORRADE_VERIFY(obb2.contains(vec3f(-5, -2, 0.5)));
+  CORRADE_VERIFY(obb2.contains(vec3f(5, 0, -0.5)));
+  CORRADE_VERIFY(!obb2.contains(vec3f(5, 0, 2)));
+  CORRADE_VERIFY(!obb2.contains(vec3f(-10, 0.5, -2)));
+
+  const box3f aabb = obb2.toAABB();
+  CORRADE_COMPARE_AS(aabb.min().x(), -10, float);
+  CORRADE_COMPARE_AS(aabb.min().y(), -5, float);
+  CORRADE_COMPARE_AS(aabb.min().z(), -1, float);
+  CORRADE_COMPARE_AS(aabb.max().x(), 10, float);
+  CORRADE_COMPARE_AS(aabb.max().y(), 5, float);
+  CORRADE_COMPARE_AS(aabb.max().z(), 1, float);
```

Comment:
```suggestion
  CORRADE_COMPARE(Mn::Vector3{aabb.min()}, (Mn::Vector3{-10.0f, -5.0f, -1.0f}));
  CORRADE_COMPARE(Mn::Vector3{aabb.max()}, (Mn::Vector3{10.0f, 5.0f, 1.0f}));
```

I.e., converting Eigen type to Magnum first and then using the convenient builtin comparisons (don't forget to [`#include <Magnum/EigenIntegration/Integration.h>`](https://doc.magnum.graphics/magnum/integration_2src_2Magnum_2EigenIntegration_2Integration_8h.html) for those). Alternatively

```cpp
  CORRADE_VERIFY(aabb.min().isApprox(vec3f{-10.0f, -5.0f, -1.0f}));
  CORRADE_VERIFY(aabb.max().isApprox(vec3f{10.0f, 5.0f, 1.0f}));
```

This is a bit of a clash between Magnum and Eigen types and I'm not sure what's better -- would the OBB class benefit from directly using Magnum types instead of Eigen? Or would that make it harder to use in certain contexts?

Reply:
I think we expose enough of OBB/AABB to python to keep it Eigen as that becomes numpy on the other end.  

My opinion is that we should keep things as eigen/numpy as much as we can as python users will expect numpy as the vector type.  Magnum types do play very well with numpy, so when there is a performance concern due to needing to transfer between numpy and magnum, we should use magnum types, but I don't think that applies here.